index.js

```diff
@@ -180,14 +187,15 @@ RedirectableRequest.prototype.setTimeout = function (msecs, callback) {
 
   // Start the timer when the socket is opened
   if (this.socket) {
-    startTimer();
+    startTimer(this.socket);
   }
   else {
     this._currentRequest.once("socket", startTimer);
   }
 
   this.once("response", clearTimer);
   this.once("error", clearTimer);
+  this.on("socket", setSocketTimeout);
```

Comment:
A side effect of this is that `setSocketTimeout()` gets called twice for the first request. We deal with that, see comment above.

Reply:
Can't we wrap this in an `if` statement then?

Reply:
Something [like this](https://github.com/follow-redirects/follow-redirects/pull/157/commits/aa476d61a2ca3b6971f12b5db98bf8cad33267e6)?

Reply:
> Can't we wrap this in an `if` statement then?

Hello, any comments on my latest commit? Thanks.